### ISSUE TITLE
[>=1.2] dev group ignored when deprecated tool.poetry.dev-dependencies is present in pyproject.toml

### ISSUE THREAD

**Issue by zEdS15B3GCwq** [python-poetry/poetry#5424]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Win 11 21H2
- **Poetry version**: 1.2.0b1 (latest as of 2022 Apr 7)

## Issue
I have a project that was created with Poetry 1.1.*, and when I upgraded Poetry to the latest dev version (1.2.0b1), the dev dependencies of the project (added in 1.1) stopped installing on `poetry install`. I've found that the deprecated section `[tool.poetry.dev-dependencies]` was still in pyproject.toml, and when it was present, poetry would not recognise the dev packages in `[tool.poetry.group.dev.dependencies]`.

Furthermore, I had some cases in which dev dependencies were removed from the project on a `poetry upgrade` and in other situations, but those cases I did not explore further so I'm just mentioning them to indicate that there may be further issues.

#### minimal example:
  ```
[tool.poetry]
name = "bb"
version = "0.1.0"
description = ""
authors = ["Your Name <you@example.com>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.10"
decorator = "^5.1.1"

[tool.poetry.group.dev.dependencies]
dill = "^0.3.4"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

Executing `poetry install` or even `poetry install --with=dev` only installs `decorator`.

If the section `[tool.poetry.dev-dependencies]` is renamed or removed, poetry will behave correctly.

I'm unsure in which situations the empty `[tool.poetry.dev-dependencies]` section can remain or appear in the project.toml file.

**Comment by abn** [comment]
This should be resolved with https://github.com/python-poetry/poetry-core/pull/305.

**Comment by zEdS15B3GCwq** [comment]
@abn , that's good to hear. Is there any chance for a fix for current 1.2?

**Comment by abn** [comment]
You could manually update poetry-core to git@master in poetry's own virtual environment. 

The next 1.2 release will have this fix.
